src/main/java/com/google/sps/data/Response.java

```diff
@@ -42,15 +42,16 @@ public String getStatus() {
     }
 
     public void pick() {
-        if(results.size() > 0){
-            int randIdx = (results.size() > 1) ? new Random().nextInt(results.size() - 1) : 0;
-            pick = results.get(randIdx);
-            results.remove(randIdx);
-        }
+        if (results.size() == 0)
+            throw new NullPointerException("results array not initialized");
```

Comment:
NullPointerException usually isn't used this way. In this case, not having `results` initialized is a programming error, so we could perhaps throw something like `IllegalStateException`.